javascript/ql/src/semmle/javascript/ES2015Modules.qll

```diff
@@ -413,6 +433,18 @@ class ExportNamedDeclaration extends ExportDeclaration, @exportnameddeclaration
   }
 }
 
+/**
+ * An export declaration with the `type` modifier.
+ */
+private class TypeOnlyExportDeclaration extends ExportNamedDeclaration {
+  TypeOnlyExportDeclaration() { isTypeOnly() }
+
+  override predicate exportsAs(LexicalName v, string name) {
+    super.exportsAs(v, name) and
+    not v instanceof Variable
```

Comment:
How about `LocalNamespaceName`? 
(I'm not sure, so if you say its fine, then its fine)

Reply:
Despite their name, type-only imports/exports also affects both types and namespaces.